### PR TITLE
Implement parsing of module definitions

### LINKED ISSUE #67: KeyError in application_program_loader.py:88
Hi,

I watend to play around with xknxproject a bit and tried to load a project.

I got this error:

```
Traceback (most recent call last):
  File "/home/weichbr/xknxproject/test.py", line 6, in <module>
    project: KNXProject = knxproj.parse()
  File "/home/weichbr/xknxproject/xknxproject/knxproj.py", line 32, in parse
    return XMLParser(knx_project_content).parse()
  File "/home/weichbr/xknxproject/xknxproject/xml/parser.py", line 40, in parse
    self.load()
  File "/home/weichbr/xknxproject/xknxproject/xml/parser.py", line 155, in load
    ApplicationProgramLoader.load(application_program_file, devices)
  File "/home/weichbr/xknxproject/xknxproject/loader/application_program_loader.py", line 88, in load
    _com_object_ref = com_object_refs[com_instance.ref_id]
KeyError: 'M-0083_A-0098-11-29A3_MD-1_M-2_MI-1_O-2-0_R-99'
```

Sadly, I cannot share the knxproj file as it contains a lot of personal information. The project is simple, IP Backbone, one TP line.

To assist a bit with debugging, this is what I could figure out:
`com_object_refs` only contains 3 items:
```
ComObjectRef(identifier='M-0083_A-0098-11-29A3_O-72_R-10', ref_id='M-0083_A-0098-11-29A3_O-72', name=None, text=None, function_text=None, object_size=None, read_flag=None, write_flag=None, communication_flag=None, transmit_flag=None, update_flag=None, read_on_init_flag=None, datapoint_type=None)
ComObjectRef(identifier='M-0083_A-0098-11-29A3_O-79_R-4', ref_id='M-0083_A-0098-11-29A3_O-79', name=None, text=None, function_text=None, object_size=None, read_flag=None, write_flag=None, communication_flag=None, transmit_flag=None, update_flag=None, read_on_init_flag=None, datapoint_type=None)
ComObjectRef(identifier='M-0083_A-0098-11-29A3_O-80_R-146', ref_id='M-0083_A-0098-11-29A3_O-80', name=None, text='Temperature', function_text='Send measured value', object_size='2 Bytes', read_flag=True, write_flag=False, communication_flag=None, transmit_flag=None, update_flag=False, read_on_init_flag=None, datapoint_type={'main': 9, 'sub': 1})
```

`com_object_instance_refs` contains 47 objects

The offending `com_instance` is number 3 in that list:
```
ComObjectInstanceRef(identifier=None, ref_id='M-0083_A-0098-11-29A3_MD-1_M-2_MI-1_O-2-0_R-99', text='T3/4: Rollladen', function_text=None, read_flag=None, write_flag=None, communication_flag=None, transmit_flag=None, update_flag=None, read_on_init_flag=None, datapoint_type=None, description=None, links=['GA-23'], name=None, number=None, object_size=None)
```

`used_com_object_refs` contains 17 strings and `M-0083_A-0098-11-29A3_MD-1_M-2_MI-1_O-2-0_R-99` __is__ one of them.

I printed all the identifiers it loops over and all are missing the `MD-?_M-?_MI-?_` part:
grafik

Yet, there are a lot of comobjrefs that have these (`used_com_object_refs`):
grafik

Please let me know what I should investigate if you need more information.

### PR DESCRIPTION
**Pull request by marvin-w**
Implement proper parsing of module definitions for KNXProj application programs.

Has a minor impact on performance due to the fact that we now need to parse more content since the structure of the module definitions are looking like this:

```xml
<ModuleDef>
  ...
     <Static>
```

and we previously stopped parsing when we saw a "Static" tag in the XML. (We don't parse languages and dynamic stuff)

Fixes #67

### CODE REVIEW

File: xknxproject/loader/application_program_loader.py [hunk 1]
```diff
@@ -78,8 +78,8 @@ def load(
                             ),
                             datapoint_type=datapoint_type,
                         )
-                if elem.tag.endswith("Static"):
-                    # we don't need anything after Static tag (Dynamic, Languages)
+                if elem.tag.endswith("Dynamic"):
+                    # We don't need anything after dynamic
```
Comment (farmio) [inline xknxproject/loader/application_program_loader.py:82]: We should stop when reaching the end of `ApplicationPrograms` currently.
`Dynamic` (which we don't need), as well as `ModuleDefs` (which we need), both are optional and come after `Static`. If we had an application without a `Dynamic` block we would parse `Languages` (translations) as well with the current implementation - which we don't need.

File: xknxproject/loader/application_program_loader.py [hunk 2]
```diff
@@ -78,8 +78,8 @@ def load(
                             ),
                             datapoint_type=datapoint_type,
                         )
-                if elem.tag.endswith("Static"):
-                    # we don't need anything after Static tag (Dynamic, Languages)
+                if elem.tag.endswith("Dynamic"):
```
Comment (marvin-w) [inline xknxproject/loader/application_program_loader.py:81]:
```suggestion
                if elem.tag.endswith("ApplicationPrograms"):
```

File: xknxproject/loader/application_program_loader.py (hunk 1 above)
Comment (marvin-w) [inline xknxproject/loader/application_program_loader.py:82]:
```suggestion
                    # We don't need anything after ApplicationPrograms
```

**Review by farmio — APPROVED** [review]
Nice! 🎉

**Pull request merged**